From 0bee992a4338007381f83b95a7f9d9346302e7e7 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Tue, 24 Nov 2015 10:13:00 +0530
Subject: [PATCH 10/12] [PMIC_CCSM]: Correct Vbus detection status register

The VBUS detection bit is in the Power Source Status Register. But
the existing code is looking from CCSM status register. Use the right
register to determine the VBUS status.

Change-Id: I377207c47e20c9eb1eb5467c520b30886b7fe549
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9715
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
Reviewed-on: https://android.intel.com:443/439699
(cherry picked from commit 9975be96c36c2043c799b860b1fab5c83ed30277)
Reviewed-on: https://android.intel.com:443/442598
---
 drivers/power/intel_pmic_ccsm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index 63d51f8..a995b84 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -1847,7 +1847,7 @@ static void pmic_ccsm_process_cable_events(enum cable_type cbl_type,
 	case CABLE_TYPE_SINK:
 		/* Check for VBUS status */
 		if (cable_state) {
-			ret = pmic_read_reg(chc.reg_map->pmic_usbpath, &val);
+			ret = pmic_read_reg(chc.reg_map->pmic_schgrirq1, &val);
 			if (ret < 0) {
 				dev_err(chc.dev,
 					"%s: Failed to read VBUS status\n",
-- 
1.9.1

